### PR TITLE
Chakra 2, React 18 and more

### PR DESCRIPTION
**Pull request by garrylachman**
So...
* Migrate to Chakra 2+
* Upgrade React 18
* Fix tests (media query mock has been added)
* Upgrade Storybook
* Upgrade Typescript
* Upgrade Babbel
* Most of the dependencies has been updated
* Config files has been changes to support the new dependecies and react 18

### CODE REVIEW

**Comment by garrylachman** [comment]
`"@babel/runtime": "7.17.7"` must be changed, the build will fail if not

**Comment by WarriorCodeDev** [comment]
@moosefaceee

**Comment by garrylachman** [comment]
found some issues, will fix soon ---- On Mon, 21 Nov 2022 15:28:09 +0200  ***@***.***  wrote ----
@moosefaceee

**Comment by jeanverster** [comment]
Hey @garrylachman  - thanks for taking the time to make these changes, much appreciated!

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -9,23 +9,6 @@ on:
...
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v1
-        with:
-          node-version: "14"
-          registry-url: "https://registry.npmjs.org"
-      - run: yarn
-
-      - run: npm config set registry https://registry.npmjs.org && npm publish
-        env:
-          NODE_AUTH_TOKEN: ${{ secrets.npm_token }}
```
[the remark this answers is not in the document]
Comment (garrylachman) [reply .github/workflows/publish.yml:28]: sorry just for my test...

[the remark this answers is not in the document]
Comment (garrylachman) [reply .github/workflows/publish.yml:28]: to run a build in my fork

**Comment by MatthewCushing** [comment]
Are you looking to review this anytime soon by chance?

**Comment by moosefaceee** [comment]
Also keen to use this on a newish v18 project

**Comment by jeanverster** [comment]
Hey all! Sorry for the lack of input here. I've recently released v2 of this component which includes support for chakra 2 and react 18. Let me know if you still run into issues! 

Gonna close this PR for now as I don't think it's necessary anymore. Thanks for the effort @garrylachman 💪

**Pull request closed without merging**